### PR TITLE
[1658] Fix schools progress on review-draft page

### PR DESCRIPTION
**Pull request by AbigailMcP**
### Context

https://trello.com/c/B0soQqjA/1658-update-schools-progress-tag-on-review-draft-page

### Changes proposed in this pull request

- Created a new form validator `Schools::FormValidator` that is responsible for validating the lead school form and the employing school form as one.
  - For `school_direct_tuition_fee` trainees, only `LeadSchoolForm` validations are run
  - For `school_direct_salaried` trainees, only `EmployingSchoolForm` validations are run
- Moved all our `School` related forms into a module.

### Guidance to review

- Visit a `school_direct_salaried` trainee, with no schools added
- Check that the 'Schools' section on `trainee/review-draft` is 'Not started'
- Add just a lead school
- Check that the 'Schools' section is 'In progress'
- Add a lead school and an employing school
- Check that the 'Schools' section is still 'In progress'
- Add a lead school and an employing school and check the 'I have completed this section' box
- Check that the 'Schools' section is 'Complete'

### CODE REVIEW

File: app/forms/schools/employing_school_form.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module Schools
```
Comment (AbigailMcP) [inline app/forms/schools/employing_school_form.rb:3]: Only change is putting this code inside a module.

File: app/forms/schools/form.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module Schools
```
Comment (AbigailMcP) [inline app/forms/schools/form.rb:3]: Only change is putting this code inside a module.

File: app/forms/schools/lead_school_form.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module Schools
```
Comment (AbigailMcP) [inline app/forms/schools/lead_school_form.rb:3]: Only change is putting this code inside a module.

File: spec/forms/schools/employing_school_form_spec.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+module Schools
```
Comment (AbigailMcP) [inline spec/forms/schools/employing_school_form_spec.rb:5]: Only change is putting this code inside a module.

File: spec/forms/schools/lead_school_form_spec.rb [hunk 5]
```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+module Schools
```
Comment (AbigailMcP) [inline spec/forms/schools/lead_school_form_spec.rb:5]: Only change is putting this code inside a module.

File: app/forms/schools/form_validator.rb [hunk 6]
```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module Schools
+  class FormValidator
```
Comment (darokel) [inline app/forms/schools/form_validator.rb:4]: 👌

**Review by darokel — APPROVED** [review]
👍

**Comment by AbigailMcP** [comment]
[Fix schools 'progress' tag on review-draft page](https://trello.com/c/B0soQqjA/1658-fix-schools-progress-tag-on-review-draft-page)

**Comment by GeoffJohnson** [comment]
All good

**Pull request merged**